tideways/provision.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+# Tideways with XHgui
+DIR=`dirname $0`
+
+install_tideways() {
+    # Tideways is only for php =>7.0
+    echo "Installing/update Tideways to PHP 7.0, 7.1, 7.2"
+    git clone "https://github.com/tideways/php-xhprof-extension" "/var/local/tideways-php7.2"
+    cp -r /var/local/tideways-php7.2 /var/local/tideways-php7.0
+    cp -r /var/local/tideways-php7.2 /var/local/tideways-php7.1
+    for version in 7.0 7.1 7.2
+        do
+        cd "/var/local/tideways-php${version}"
+        update-alternatives --set php /usr/bin/php$version
+        update-alternatives --set php-config /usr/bin/php-config$version
+        update-alternatives --set phpize /usr/bin/phpize$version
+        phpize$version
+        ./configure --enable-tideways-xhprof --with-php-config=php-config$version
+        make
+        make install
+    done
+}
+
+echo "Installing Tideways & XHgui"
+if [[ ! -d "/srv/www/default/xhgui" ]]; then
+    if [[ -d "/etc/php/5.6/" ]]; then
+        echo "File copied for php 5.6"
+        cp "${DIR}/tideways.ini" "/etc/php/5.6/mods-available/tideways_xhprof.ini"
+        cp "${DIR}/mongodb.ini" "/etc/php/5.6/mods-available/mongodb.ini"
+    fi
+    if [[ -d "/etc/php/7.0/" ]]; then
+        echo "File copied for php 7.0"
+        cp "${DIR}/tideways.ini" "/etc/php/7.0/mods-available/tideways_xhprof.ini"
+        cp "${DIR}/mongodb.ini" "/etc/php/7.0/mods-available/mongodb.ini"
+        cp "${DIR}/xhgui-php.ini" "/etc/php/7.0/fpm/conf.d/xhgui.ini"
+    fi
+    if [[ -d "/etc/php/7.1/" ]]; then
+        echo "File copied for php 7.1"
+        cp "${DIR}/tideways.ini" "/etc/php/7.1/mods-available/tideways_xhprof.ini"
+        cp "${DIR}/mongodb.ini" "/etc/php/7.1/mods-available/mongodb.ini"
+        cp "${DIR}/xhgui-php.ini" "/etc/php/7.1/fpm/conf.d/xhgui.ini"
+    fi
+    if [[ -d "/etc/php/7.2/" ]]; then
+        echo "File copied for php 7.2"
+        cp "${DIR}/tideways.ini" "/etc/php/7.2/mods-available/tideways_xhprof.ini"
+        cp "${DIR}/xhgui-php.ini" "/etc/php/7.2/fpm/conf.d/xhgui.ini"
+        # For the default php version
+        cp "${DIR}/mongodb.ini" "/etc/php/7.2/mods-available/mongodb.ini"
+    fi
+    apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 0C49F3730359A14518585931BC711F9BA15703C6
+    echo "deb http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-3.4.list
+    sudo apt update
+    apt-get -y install mongodb-org re2c
```

Comment:
Does it make sense to add mongodb as a separate utility option? It should probably be separated into its own function at least

Reply:
Do you mean move in a function inside this script?

Reply:
That's right

Reply:
I think that is one is done